### DISCUSSION TITLE
Synchronous Multi agent

### DISCUSSION THREAD

**Discussion by Rakesh-123-cryp** [mesa/mesa#2076] [Ideas]
And how about a synchronous execution for multi-agent environment. The several agents in the environment line up with jobs that are executed in parallel with no time lag. Multi agent environment are tricky to handle and the execution performed on the by different agents at the same time must overlap. If agent one takes more time to decide the action and agents two takes less the parallelism, if applicable, can we used to speed up the first agent.

**Comment by rht** [comment]
Past discussion on parallelization within a single simulation: https://github.com/projectmesa/mesa/discussions/1981#discussioncomment-8196148.
